README.md

```diff
@@ -677,17 +677,64 @@ In either case:
     `{...}`, but they should ask themselves if this code is really readable and
     whether the block's content can be extracted into nifty methods.
 
-* Avoid `return` where not required.
+* Use `return` even if it is not required in Ruby language.
+
+    Although it is a Ruby way that an explicit return is unnecessary, it introduces
+    confusions to programmers working with unfamilar code and to programmers learning
+    Ruby from other strict languages.
 
     ```Ruby
-    # bad
+    # bad: it is unclear whether this function want to return the updated value
+    # or just want to set the new value
+    def update(new_value)
```

Comment:
I would argue that the amendment here should be something along the lines of 'when you don't wish to return the last value from a method, return `nil` instead'. That way it's clear when someone doesn't desire to return the value of an assignment.